packages/editor/src/components/editor/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { TEMPLATE_POST_TYPE } from '../../store/constants';
+import EditorInterface from '../editor-interface';
+import EditorProvider from '../provider';
+import Sidebar from '../sidebar';
+
+function Editor( {
+	postType,
+	postId,
+	templateId,
+	settings,
+	children,
+
+	// This could be part of the settings.
+	onActionPerformed,
+
+	// The following abstractions are not ideal but necessary
+	// to account for site editor and post editor differences for now.
+	className,
+	styles,
+	customSaveButton,
+	forceDisableBlockTools,
+	title,
+	iframeProps,
+	extraSidebarPanels,
+	enableRegionNavigation = true,
+} ) {
+	const { post, template, hasLoadedPost } = useSelect(
+		( select ) => {
+			const { getEntityRecord, hasFinishedResolution } =
+				select( coreStore );
+			return {
+				post: getEntityRecord( 'postType', postType, postId ),
+				template: templateId
+					? getEntityRecord(
+							'postType',
+							TEMPLATE_POST_TYPE,
+							templateId
+					  )
+					: undefined,
+				hasLoadedPost: hasFinishedResolution( 'getEntityRecord', [
+					'postType',
+					postType,
+					postId,
+				] ),
+			};
+		},
+		[ postType, postId, templateId ]
+	);
+
+	return (
+		<EditorProvider
+			post={ post }
+			__unstableTemplate={ template }
```

Comment:
To be honest, I don't really think that this EditorProvider should receive the full objects, postType, postId should suffice but it's a very old component and I didn't want to touch its API.